shm_wrapper/shm_wrapper.h

```diff
@@ -45,17 +49,26 @@ typedef struct {
     sem_t* command_sem;  // semaphore on the command stream of a device
 } dual_sem_t;
 
-// shared memory for gamepad
+
+// struct describing an input
 typedef struct {
-    uint32_t buttons;    // bitmap for which buttons are pressed
-    float joysticks[4];  // array to hold joystick positions
-} gamepad_shm_t;
+    uint64_t buttons;           // bitmap for which buttons are pressed
+    float joysticks[4];         // array to hold joystick positions, only for gamepad
+    robot_desc_field_t source;  // which hardware device the command came from, either GAMEPAD or KEYBOARD
+} input_t;
+
+// shared memory for gamepad and keyboard inputs
+typedef struct {
+    input_t inputs[2];  // Index 0 is for GAMEPAD, index 1 is for KEYBOARD, like in the enum. can be modified in the future
```

Comment:
I mentioned this in another comment but you can use the `#define` here for the 2